### PR TITLE
build(ci): Fix merge base for visual snapshots

### PR DESCRIPTION
**Pull request by billyvg**
When we moved from the workflow trigger event `pull_request` to `pull_request_target` (which we changed to allow forks to run Visual Snapshots), it resulted in a change of the pull request base SHA. Instead of relying on the github context for the merge base, we should use `git merge-base` to find the base sha.

The merge base SHA is then used to use a three-way comparison for PRs (comparing the PR head SHA against latest master and the merge base).

### CODE REVIEW

**Comment by billyvg** [comment]
`pull_request_target` is annoying to test - you need to create a PR with your feature branch as the base (because it will only use the workflow from the base branch/master). Here is my testing branch https://github.com/getsentry/sentry/pull/22101 which uses a fork that is quite outdated (around a week or more). The number of snapshot changes is misleading as it doesn't quite detect the change of width as not important. But the tests prefix had 471 changes (vs the 150 or so here).

File: .github/workflows/acceptance.yml [hunk 1]
```diff
@@ -181,10 +181,29 @@ jobs:
...
+        if: github.event.pull_request.head.ref != ''
+        id: merge-base
+        run: |
+          echo "::set-output name=sha::$(git merge-base origin/master ${{ github.event.pull_request.head.ref }})"
+
       - name: Diff snapshots
         id: visual-snapshots-diff
         uses: getsentry/action-visual-snapshot@v2
         with:
+          merge-base: ${{ steps.merge-base.outputs.sha }}
```
Comment (joshuarli) [inline .github/workflows/acceptance.yml:206]: note: intro'd in https://github.com/getsentry/action-visual-snapshot/pull/21

**Review by armenzg — COMMENTED** [review]
Is this to improve on this task? https://app.asana.com/0/1197393184799586/1199078519540749/f
Do we make reference to Asana tasks in our commit messages?

I'm looking at the [visual snapshot results for this PR](https://github.com/getsentry/sentry/pull/22111/checks?check_run_id=1415806505) and it shows the "stack traces" -> "stacktraces" pollution. Does this PR intend to fully solve the problem? or is it a step towards it?

**Comment by billyvg** [comment]
@armenzg Yeah sorry, I should have linked that Asana ticket -- see [this comment](https://github.com/getsentry/sentry/pull/22111#issuecomment-729330675) for why this PR's snapshots show the stack traces. tldr, it runs the workflow on the base branch for a PR (in this case, `master`). So snapshots for this PR are not using the changes from this PR.

**Review by armenzg — APPROVED** [review]

**Comment by armenzg** [comment]
Thanks for clarifying!

**Pull request merged**